### PR TITLE
[proof-new] Making the CDCL(T) Minisat proof producing

### PR DESCRIPTION
**Pull request by HanielB**
This closely follows the old proof code in terms of where Minisat is hooked to the SatProofManager, other than a few places like registering removed clauses and removal of redundant literals. Note that this together with the thorough handling from SatProofManager makes the new SAT proofs perceptibly more robust than the old ones.

This PR also adds some traces to better track what Minisat does.

### CODE REVIEW

File: src/prop/minisat/core/Solver.cc [hunk 1]
```diff
@@ -920,9 +1036,17 @@ int Solver::analyze(CRef confl, vec<Lit>& out_learnt, int& out_btlevel)
             }
 
             // FIXME: can we do it lazily if we actually need the proof?
-            if (options::unsatCores() && level(var(q)) == 0)
+            if ((options::unsatCores() || isProofEnabled())
```
Comment (ajreynol) [inline src/prop/minisat/core/Solver.cc:1039]: Maybe drop the options check here, analogous to what you did on line 1061.

File: src/prop/minisat/core/Solver.cc [hunk 2]
```diff
@@ -934,13 +1058,30 @@ int Solver::analyze(CRef confl, vec<Lit>& out_learnt, int& out_btlevel)
...
+          }
+          if (isProofEnabled())
+          {
+            d_pfManager->addResolutionStep(ca[confl], p);
+          }
         }
 
     }while (pathC > 0);
     out_learnt[0] = ~p;
+    if (CVC4::options::proofNew())
```
Comment (ajreynol) [inline src/prop/minisat/core/Solver.cc:1075]:
```suggestion
    if (options::proofNew())
```

Comment (ajreynol) [reply src/prop/minisat/core/Solver.cc:1075]: Should this just be guarded with `Debug.isOn("newproof::sat")`?

**Review by ajreynol — CHANGES_REQUESTED** [review]
A few minor things

File: src/prop/minisat/core/Solver.cc (hunk 2 above)
Comment (HanielB) [reply src/prop/minisat/core/Solver.cc:1075]: Yes :)

**Review by ajreynol — APPROVED** [review]

**Pull request merged**